### ISSUE TITLE
OVH: Long DKIM records not working

### ISSUE THREAD

**Issue by patschi** [StackExchange/dnscontrol#487]
Unfortunately using (very) long DKIM keys with `TXT()` does not seem to work properly with OVH. MultiTXT does clearly not work (even when enabling it temporarily during debugging) as OVH simply rejects it.

**The situation**: In general the DKIM is fine in a TXT. However when using `rtype=TXT`, OVH has a specific limit. When using `rtype=DKIM` however, it works just fine due to a longer limit. Both mentioned record types are basically code-wise identical.

**The workaround:**
The only failing operations are CREATE and MODIFY, therefore it might be enough to simply re-write the `FieldType`/`rc.Type` to `DKIM` just before the API request is sent. This works and updates the long DKIM without any issues. (Tested by overwritting it while debugging)

I'm just failing to find a good way to determine if the actual `TXT` record, which is going to be updated, is a DKIM entry... Doing a `strings.Contains()` on the `Target` doesn't sound like a good way for me.

Any thoughts? Any help/fix is greatly appreciated! Thanks in advance! `DKIM` is so an important record for me which just fails updating every time.

**Relevant code parts:**
https://github.com/StackExchange/dnscontrol/blob/a7eba97ada193e7ed72be1df9dd87cbbbd6642e7/providers/ovh/protocol.go#L112-L122

https://github.com/StackExchange/dnscontrol/blob/a7eba97ada193e7ed72be1df9dd87cbbbd6642e7/providers/ovh/protocol.go#L130-L142

**Comment by captncraig** [comment]
Do they translate the DKIM records to TXT under the covers? That seems wrong once there is already a native DKIM record that is essentially deprecated.

**Comment by patschi** [comment]
Yes, they do translate it to TXT afterwards, just like they do with the SPF record.

I just thought of two ways:
1. Checking if the TXT value contains specific patterns which matches DKIM.
2. Provide a modifier like `IS_DKIM` so it can handle it slightly differently based on the provider's needs.

**Comment by patschi** [comment]
Just to clarify further with an example:
```go
 func (c *ovhProvider) updateRecordFunc(old *Record, rc *models.RecordConfig, fqdn string) func() error { 
 	return func() error {
 		rc.Type = "DKIM" // <-------- THAT WORKS
 		record := Record{ 
 			SubDomain: rc.GetLabel(), 
 			FieldType: rc.Type, 
 			Target:    rc.GetTargetCombined(), 
 			TTL:       rc.TTL, 
 			Zone:      fqdn, 
 			ID:        old.ID, 
 		} 
```

I've checked the `models.RecordConfig` struct, and there's a `Metadata` string array. My first idea was simply using `if rc.Metadata["IS_DKIM"] == "true" {}` to get the DKIM-workaround applied (without adding a new RTYPE, or new struct variable). But I'm unsure if...
1) is possible to pass such a value from the JS-definition to the Metadata array, 
2) if that's the most "elegant" way doing this.

**Comment by captncraig** [comment]
That is... a really frustrating api choice. I don't think it is a good idea to change the front-end inputs to accommodate their bad design. One of our main goals is for the input to be as universal as possible, and the individual drivers handle all of the oddities and intricacies of each provider.

The ovh provider could make a loop through all TXT records, and convert dkim and/or spf records into the appropriate provider-specific types as necessary. I would hope looking for `v=DKIM1`, or maybe checking name for `_domainkey` would be an ok way to identify them, but not sure because examples in the wild are all over the place.

**Comment by patschi** [comment]
I've filed a PR #488 with as few code changes as possible and this way it works just fine with longer DKIM keys with OVH. I can't change the record type earlier to `DKIM` as it may throw errors that there's no valid known DKIM rType defined in the code, and also not to confuse the `domainCorrections`-stuff.

**Comment by patschi** [comment]
Fixed with #488.
